### PR TITLE
[FixDanishLetterI] - Refact + Optimization for FixDanishLetterI.

### PR DESCRIPTION
**Pull request by ivandrofly**
Fix for https://github.com/SubtitleEdit/subtitleedit/pull/978.

I left some notes if you guys can help would be nice :). /cc @niksedk @xylographe 

Clone this branch:
`git fetch https://github.com/ivandrofly/subtitleedit.git danishI:localDanishI`

Can be merged /cc @niksedk

### CODE REVIEW

File: libse/SubtitleEditRegex.cs [hunk 1]
```diff
@@ -94,14 +94,13 @@ static DanishLetterI()
...
                     RegexFactory("\b[Kk]ender i (?:hinanden|to hinanden)\b"),
-                    RegexFactory("\b[Kk]endte i(?: | hinanden)\b"), // Review
+                    RegexFactory("\b[Kk]endte i (?: hinanden)?\b"),
                     RegexFactory("\b[Kk](?:iggede|igger) i på"),
                     RegexFactory("\b[Kk]ommer i (?:her|ofte|sammen|tit)\b"),
                     RegexFactory("\b[Kk]unne i (?:fortælle|give|gøre|ikke|lide|mødes|se)\b"),
                     RegexFactory("\b[Ll]eder i efter\b"),
                     RegexFactory("\b[Ll]aver i (?:ikke|her)\b"),
                     RegexFactory("\b[Ll]igner i (?:far|hinanden|mor)\b"),
-                    RegexFactory("\b[Ll]eder i efter\b"),
```
Comment (ivandrofly) [inline libse/SubtitleEditRegex.cs]: Duplicated

File: libse/Forms/FixCommonErrors/FixDanishLetterI.cs [hunk 2]
```diff
@@ -17,626 +17,19 @@ public void Fix(Subtitle subtitle, IFixCallbacks callbacks)
...
-                                 MyRegEx(@"\ber i sultne\b"),
-                                 MyRegEx(@"\bEr i tilfredse\b"),
-                                 MyRegEx(@"\ber i tilfredse\b"),
-                                 MyRegEx(@"\bEr i to\b"),
-                                 MyRegEx(@"\ber i ved at\b"),
-                                 MyRegEx(@"\ber i virkelig\b"),
-                                 MyRegEx(@"\bEr i virkelig\b"),
-                                 MyRegEx(@"\bEr i vågne\b"),
-                                 MyRegEx(@"\ber i vågne\b"),
-                                 MyRegEx(@"\bfanden vil i?"),
```
Comment (ivandrofly) [inline libse/Forms/FixCommonErrors/FixDanishLetterI.cs]: The intention is to match like: `i\?` right? /cc @niksedk

Comment (niksedk) [inline libse/Forms/FixCommonErrors/FixDanishLetterI.cs]: It should be the ending of a question like "What do you want?"... so it looks like a bug.

Comment (ivandrofly) [inline libse/Forms/FixCommonErrors/FixDanishLetterI.cs]: > It should be the ending of a question like "What do you want?"... so it looks like a bug.

yeah, should be `\?` instead :)

**Comment by ivandrofly** [comment]
Can be merged /cc @niksedk

**Pull request merged**